operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java

```diff
@@ -64,13 +89,36 @@ private void stopLeading() {
   }
 
   private String identity(LeaderElectionConfiguration config) {
-    String id = config.getIdentity().orElse(System.getenv("HOSTNAME"));
+    var id = config.getIdentity().orElse(System.getenv("HOSTNAME"));
     if (id == null || id.isBlank()) {
       id = UUID.randomUUID().toString();
     }
     return id;
   }
 
+  private static Optional<String> tryNamespaceFromPath() {
```

Comment:
Not sure if this is needed, could you try instead just use the client's facilities, like:
`Config.autoConfigure(null).getNamespace()` ?

Reply:
🤔 Would this lead to the same NS? And actually, we should probably get it from `ConfigurationServiceProvider.instance().getClientConfiguration()` instead of creating a new `Config` object that might not match the configured one.

Reply:
```java
default Config getClientConfiguration() {
  return Config.autoConfigure(null);
}
```
This seems create a new config as well.

I can give it a try and see whether those tests still work.

Reply:
I could get tests pass by setting `System.setProperty(KUBERNETES_AUTH_TRYKUBECONFIG_SYSTEM_PROPERTY, "false");`, but I worry that loading a full config here might not be a correct approach because it confuses users if they do not expect `.kube/config` to be read at all. The intention of this change is to only make it work in-cluster.

Reply:
Good point, maybe we can then just stick with the current impl for now

Reply:
> This seems create a new config as well.

That's just the default implementation, though. The Quarkus extension, for example, returns the configuration associated with the client (i.e. the one the client uses to connect to the cluster). 

Are there use cases where the config would be configured to use a different namespace than the one associated with the service account? Do we specifically only want the SA namespace regardless of how the Kubernetes config is set up?

Reply:
@metacosm OK I see what you mean now. Let me try something.

Reply:
So something like https://github.com/java-operator-sdk/java-operator-sdk/pull/1450/commits/0c1294753dba5cdca429e8f1487f792b4cf62bb3 ? If the config was loaded from `.kube/config`, there is for sure no namespace; if it is in-cluster config, the namespace will be read from `/var/run/secrets/kubernetes.io/serviceaccount/namespace` if no overridden.